presets/ragengine/requirements.txt

```diff
@@ -7,12 +7,31 @@ llama-index-embeddings-huggingface-api
 llama-index-llms-huggingface
 llama-index-llms-huggingface-api
 
-fastapi
-faiss-cpu
+fastapi>=0.100.0
+faiss-cpu>=1.7.4
 llama-index-vector-stores-faiss
 llama-index-vector-stores-chroma
 llama-index-vector-stores-azurecosmosmongo
-uvicorn
+uvicorn>=0.22.0
 asyncio
 aiorwlock
 httpx==0.27.0
+# Metrics collection
+prometheus-client>=0.17.0
+# Async HTTP client
+aiohttp>=3.8.0
+
+# Core dependencies
+numpy>=1.24.0
+pydantic>=2.0.0
+
+# Embedding dependencies
+sentence-transformers>=2.2.2
+torch>=2.0.0
+transformers>=4.30.0
+
+# Vector store dependencies
+scikit-learn>=1.2.0
+
+# Async HTTP client
+aiohttp>=3.8.0
```

Comment:
I don't think these highlighted libraries are needed? From the brief check I did I couldn't find any references in ragengine. Remember introducing more libraries means more breaking changes in future.